pkg/transformer/kubernetes/kubernetes_test.go

```diff
@@ -451,3 +453,39 @@ func TestUnsupportedKeys(t *testing.T) {
 	}
 
 }
+
+func TestRestartOnFailure(t *testing.T) {
+
+	kobjectWithRestartOnFailure := newKomposeObject()
+	serviceConfig := kobjectWithRestartOnFailure.ServiceConfigs["app"]
+	serviceConfig.Restart = "on-failure"
+	kobjectWithRestartOnFailure.ServiceConfigs = map[string]kobject.ServiceConfig{"app": serviceConfig}
+
+	// define all test cases for RestartOnFailure function
+	replicas := 2
+	testCase := map[string]struct {
+		komposeObject kobject.KomposeObject
+		opt           kobject.ConvertOptions
+	}{
+		// objects generated are deployment, service and replication controller
+		"Do not Create Deployment (D) with restart:'on-failure'":             {kobjectWithRestartOnFailure, kobject.ConvertOptions{IsDeploymentFlag: true, Replicas: replicas}},
+		"Do not Create DaemonSet (DS) with restart:'on-failure'":             {kobjectWithRestartOnFailure, kobject.ConvertOptions{IsDaemonSetFlag: true, Replicas: replicas}},
+		"Do not Create ReplicationController (RC) with restart:'on-failure'": {kobjectWithRestartOnFailure, kobject.ConvertOptions{IsReplicationControllerFlag: true, Replicas: replicas}},
+	}
+
+	for name, test := range testCase {
+		t.Log("Test case:", name)
+		k := Kubernetes{}
+		if os.Getenv("BE_CRASHER") == "1" {
```

Comment:
Here we are testing a function which results in `logus.Fatalf()` when a condition is met, which further call `os.Exit()` and exits the process. If we write a test in the usual way that will call the function, it will exit the process and the running process is actually the test process and our test would fail. 
So to test the function resulting in `os.Exit()` we need  invoke `go test` again in a separate process through `exec.Command`, limiting execution to the `TestRestartOnFailure` test using  `-test.run=TestRestartOnFailure` flag set.

The `TestRestartOnFailure` is two things simultaneously, it is going to the be the test itself and second it will a be `subprocess` that the test runs.
`cmd := exec.Command(os.Args[0], "-test.run=TestRestartOnFailure")` will execute the test binary with the flag `-test.run=TestRestartOnFailure` and set the environment variable `BE_CRASHER=1`.
`err := cmd.Run()` will re-execute the test binary and this time `os.Getenv("BE_CRASHER") == "1"` will return true and we can call `o.Transform(test.komposeObject, test.opt)`.
So now we are having a test binary that calls itself and execute the code on behalf of the parent process. 
[Refence](https://talks.golang.org/2014/testing.slide#23)